driven/storage/adapter.go

```diff
@@ -342,6 +342,40 @@ func filterArgs(filter bson.M) *logutils.FieldArgs {
 	return &args
 }
 
+// FindScores finds scores based on filter
+func (a *Adapter) FindScores(filter bson.M, limit *int, offset *int) ([]model.Score, error) {
+	var scores []model.Score
+	err := a.db.scores.Find(a.context, filter, &scores, nil)
+	if err != nil {
+		return nil, errors.WrapErrorAction(logutils.ActionFind, model.TypeScore, filterArgs(filter), err)
+	}
+
+	return scores, nil
+}
+
+// UpdateScoreByFilter updates a score based on filter
+func (a *Adapter) UpdateScoreByFilter(filter bson.M, update bson.M) error {
+	_, err := a.db.scores.UpdateOne(a.context, filter, update, nil)
+	if err != nil {
+		return errors.WrapErrorAction(logutils.ActionUpdate, model.TypeScore, filterArgs(filter), err)
+	}
+
+	return nil
+}
```

Comment:
Your bson should be constructed inside these two functions, not passed in. This will allow you to avoid importing mongo dependencies outside the storage package.